pallets/messages/src/lib.rs

```diff
@@ -350,11 +350,16 @@ impl<T: Config> Pallet<T> {
 		schema_id: SchemaId,
 		schema_payload_location: PayloadLocation,
 		block_number: T::BlockNumber,
-	) -> Vec<MessageResponse<T::BlockNumber>> {
+	) -> Vec<MessageResponse> {
 		<Messages<T>>::get(block_number, schema_id)
 			.into_inner()
 			.iter()
-			.map(|msg| msg.map_to_response(block_number, schema_payload_location))
+			.map(|msg| {
+				msg.map_to_response(
+					block_number.try_into().unwrap_or_default(),
```

Comment:
Worried about unwrap or default being in the runtime. I wonder if this map function should be on the client side. Shouldn't change in this PR.

I know it is "impossible" for unwrap_or_default here to fail, but it is hiding an overflow error.

I've got an idea. Perhaps instead we move this out of the iter and do a check at the top and return an empty vec if that happens?

Really we should just move this mapping to the client (where it can also par_iter as this is just a mapping).

Thoughts?

Reply:
Good idea just resolve it before the call and error if overflow

Good catch

Reply:
@wilwade  I have resolved the number outside however this is a tricky place, it will always resolve to default if error occurs which keep us safe

Still i moved it out